src/layouts/Settings/PrivacySettings.tsx

```diff
@@ -0,0 +1,173 @@
+import {
+  VStack,
+  FormControl,
+  Box,
+  Input,
+  Flex,
+  InputGroup,
+  Icon,
+  InputRightElement,
+  Popover,
+  PopoverTrigger,
+  PopoverContent,
+  PopoverArrow,
+  PopoverBody,
+  Text,
+} from "@chakra-ui/react"
+import {
+  FormErrorMessage,
+  FormLabel,
+  IconButton,
+} from "@opengovsg/design-system-react"
+import { useEffect, useState } from "react"
+import { useFormContext, useFormState } from "react-hook-form"
+import { BiCopy } from "react-icons/bi"
+
+import { Section, SectionHeader } from "layouts/components"
+
+import { generatePassword } from "utils/password"
+import { PASSWORD_REGEX } from "utils/validators"
+
+import { SitePasswordSettings } from "types/settings"
+
+import { FormToggle } from "./components/FormToggle"
+
+interface PrivacySettingsProp {
+  isError: boolean
+}
+
+export const PrivacySettings = ({
+  isError,
+}: PrivacySettingsProp): JSX.Element => {
+  const { register, trigger, watch, setValue, getValues } = useFormContext()
+  const { errors } = useFormState<SitePasswordSettings>()
+  const [hasCopied, setHasCopied] = useState(false)
+  const [isBuffering, setIsBuffering] = useState(false)
+
+  const generateAcceptablePassword = () => {
+    const regex = new RegExp(PASSWORD_REGEX)
+    let password = ""
+    while (!regex.test(password)) {
```

Comment:
would it be easier to just use a password generation lib ? https://www.npmjs.com/package/generate-password

Reply:
Yup, that makes a lot of sense, have swapped it out in https://github.com/isomerpages/isomercms-frontend/pull/1316/commits/9463fa9a4c9d7791f8d8b41935cd03ef5f111a01